service/src/test/java/bio/terra/workspace/pact/TpsApiTest.java

```diff
@@ -132,17 +132,18 @@ public void testCreatingAPolicyWithNoExistingPolicy() throws Exception {
   }
 
   @Pact(consumer = "wsm", provider = "tps")
-  public RequestResponsePact createPaoWithAPreexistingPolicy(PactDslWithProvider builder) {
+  public V4Pact createPaoWithAPreexistingPolicy(PactBuilder builder) {
     return builder
-        .given(existingPolicyState)
+        .usingLegacyDsl()
+        .given("a policy with the given id exists", Map.of("id", existingPolicyId.toString()))
```

Comment:
This is an important change that I made throughout.

Several tests were _expecting_ the provider to return the specific `existingPolicyId` and failing, since the newer version of the pact JVM deliberately randomizes this stuff unless explicitly told otherwise.

So, in order for me to not to have to rewrite those expectations, this provides a mechanism for the consumer test to communicate to the provider test the exact ID it's expecting to exist.  The provider can access this using the optional `Map` arg that is provided to the `@State` annotated method that will eventually be implemented for this, and set up its stubbed/mocked responses accordingly to match the consumer expectations.

Reply:
Note: the provider tests don't exist yet, which is why I'm able to make this change without a corresponding provider fix (see https://broadinstitute.slack.com/archives/C02LMB1PQ6R/p1698849691034179?thread_ts=1698761219.655949&cid=C02LMB1PQ6R).

Reply:
The map passed in `given` is specifying values the pact expects to exist with a particular provider state.

It allows _more_ flexibility on the provider side, because it's requiring that the value matches what the provider says it is on the provider side, not an exact value.
In other words, the provider and consumer only have to agree on the names of things, not their exact value.

So if you have defined in your tests `existingPolicyId` as `some_id`:

`.path("/api/policy/v1alpha1/pao/%s".formatted(existingPolicyId))`

will generate a pact with the path as:
`"/api/policy/v1alpha1/pao/some_id"`

where: 
``` 
.pathFromProviderState(
            "/api/policy/v1alpha1/pao/%s".formatted(existingPolicyProviderStateValue),
            "/api/policy/v1alpha1/pao/%s".formatted(existingPolicyId))
```
will generate a pact with a path:
`"/api/policy/v1alpha1/pao/{{existingPolicyId}}"`

Then the provider can set `existingPolicyId` to whatever they want on their side, so long as it matches the expected type (string, uuid, etc).

I think pact might also run through some randomly generated matching values on the consumer side, to ensure your code can handle any value of the type you specified.

It's plausible that the correct way to do this changed with the version update, but the pattern shouldn't need to be changed.

Reply:
I put things back the way they were (for the most part).  One [stubborn test](https://github.com/DataBiosphere/terra-workspace-manager/pull/1517#discussion_r1378911452) resisted, and I had to omit the `valueFromProviderState` calls for it.